python/pyspark/sql/tests/test_functions.py

```diff
@@ -959,6 +961,25 @@ def test_lit_day_time_interval(self):
         actual = self.spark.range(1).select(lit(td)).first()[0]
         self.assertEqual(actual, td)
 
+    def test_np_scalars(self):
+        from pyspark.sql.functions import array_contains, array_position
+
+        df = self.spark.createDataFrame([([1, 2, 3],), ([],)], ["data"])
+        for dtype in [np.int8, np.int16, np.int32, np.int64]:
+            self.assertEqual(df.select(lit(dtype(1))).dtypes, [("1", "int")])
+            res = df.select(array_contains(df.data, dtype(1)).alias("b")).collect()
+            self.assertEqual([Row(b=True), Row(b=False)], res)
+            res = df.select(array_position(df.data, dtype(1)).alias("c")).collect()
+            self.assertEqual([Row(c=1), Row(c=0)], res)
+
+        df = self.spark.createDataFrame([([1.0, 2.0, 3.0],), ([],)], ["data"])
+        for dtype in [np.float32, np.float64]:
+            self.assertEqual(df.select(lit(dtype(1))).dtypes, [("1.0", "double")])
```

Comment:
Probably we should see how list works.
IIRC, it works as a ArrayList but we would have to convert it into an array.

We can do this separately too.

Reply:
cc @WeichenXu123 and @mengxr FYI

Reply:
Sounds good! Created [SPARK-39405](https://issues.apache.org/jira/browse/SPARK-39405) as a follow-up for that.